WooCommerce rebranding rollout

Konamiman opened the pull request:
### Changes proposed in this Pull Request:

This pull request rolls out the new WooCommerce branding for everybody, removing the feature flag that was in use during development and incorporating some other "last minute" changes that didn't use the flag.

What follows is a list of the rebranding related pull requests that this one consolidates. The first name next to each is the pull request autor, the other(s) are the reviewer(s):

#### Pull requests incorporating changes behind a feature flag

For these, the feature flag usage has been removed manually.

* https://github.com/woocommerce/woocommerce/pull/53308 - @Konamiman, @naman03malhotra 
* https://github.com/woocommerce/woocommerce/pull/53699 - @Konamiman, @naman03malhotra 
* https://github.com/woocommerce/woocommerce/pull/53716 - @Konamiman, @barryhughes 
* https://github.com/woocommerce/woocommerce/pull/54096 - @Konamiman, @jorgeatorres 
* https://github.com/woocommerce/woocommerce/pull/54378 - @andfinally, @KokkieH
* https://github.com/woocommerce/woocommerce/pull/53429 - @ilyasfoo, @adrianduffell, @psealock 
* https://github.com/woocommerce/woocommerce/pull/54147 - @Aljullu, @roykho 

#### Pull requests incorporating changes without a feature flag

For these, the changes have been incorporated either manually or via cherry-picking, except when the changes were already in the `release/9.6` branch.

* https://github.com/woocommerce/woocommerce/pull/53513 - @KokkieH, @mcliwanow
* https://github.com/woocommerce/woocommerce/pull/54775 - @andfinally, @kdevnel 
* https://github.com/woocommerce/woocommerce/pull/54751 - @ianlin, @layoutd 
* https://github.com/woocommerce/woocommerce/pull/54571 - @oaratovskyi, @elazzabi, @mordeth 

#### Additional change added directly to this pull request

A data migration is added to update the email base color setting (the actual configured color, not the default color - that one was changed in https://github.com/woocommerce/woocommerce/pull/53308), but only if the user had configured a custom color (that is, any color different from the old default color, `#7f54b3`).


### How to test the changes in this Pull Request:

Follow the testing instructions of the pull requests listed above, except that it's no longer necessary to set any feature flag.

If you are in the reviewers list of this pull request you are expected to test, at least, the one(s) for which you were the author or the reviewer; but of course, any additional testing will be welcome.

#### Testing the migration of the email base color

(This part of the review/testing is assigned to @albarin)

1. Create a testing site (Jurassic Ninja is fine) with any version of WooCommerce (9.6.0 or older). Complete or skip the setup wizard.
2. Go to _WooCommerce - Settings - Emails_, scroll down to the "Base color" setting, both the default value mentioned in the tooltip and the actual current value of the setting should be `#7f54b3`.
3. Now you need to create a zip file from a slightly tweaked version of the branch of this pull request:
    1. Modify the header of `woocommerce.php` so that the value of `Version:` in the header is `9.6.1`.
    2. Modify `class-woocommerce.php` so that the value of `$version` is `9.6.1` too.
    3. Create the zip file by running `pnpm run build:zip` from the `plugins/woocommerce` directory of the repository.
4. Update WooCommerce in your testing site by uploading the zip you just generated. 
5. Go to the WooCommerce settings page. If you see a "WooCommerce database update required" banner, click the update button, then reload the page until it says that the update is complete (or click the button to view the progress, then manually run the update action).
6. Go to _WooCommerce - Settings - Emails_, scroll down to the "Base color" setting, and now both the default and the actual value of the setting should be `#720eec`.
7. *Either* create a new testing site with a fresh install of WooCommerce (again, 9.6.0 or older), *or*: uninstall WooCommerce in your current site, run the following in the command line, and then reinstall it:

```
wp option delete woocommerce_version
wp option delete woocommerce_email_base_color
```

8. Go to _WooCommerce - Settings - Emails_, scroll down to the "Base color" setting and change it to anything, for example `#aabbcc`.
9. Repeat steps 4 and 5.
10. Go to _WooCommerce - Settings - Emails_ and scroll down to the "Base color" setting: this time, the color that you had set in step 8 should have been preserved.

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>
Prep release/9.6 for 9.6 cycle with version bump to 9.6.0

andfinally committed:
Cherrypick additional missing commit from 54775

oaratovskyi committed:
Cherrypick of the changes in 54571

ianlin committed:
Cherrypick of the changes in 54751

Konamiman committed:
Remove the WooCommerce branding feature flag.

This causes the changes in the following pull requests
to be visible inconditionally:

53308
53699
53716
54096
54378
53429
54147

It also manually incorporates the changes from 54775

Konamiman committed:
Add changelog files

andfinally reviewed (commented):
Hi @Konamiman, we're missing the changes from https://github.com/woocommerce/woocommerce/pull/54775/commits/de13d0f0f855afbb62aa19c55f407ec164ac37c4 of https://github.com/woocommerce/woocommerce/pull/54775 – changing the logo in Storybook. It affects three files in `tools/storybook`. The other changes from that PR look and test OK. Thanks!

barryhughes reviewed (approved):
In relation to:

> https://github.com/woocommerce/woocommerce/pull/53716 - @Konamiman, @barryhughes

I see the new logo etc in the following spots:

- [x] Android app banner
- [x] Download mobile app modal

LGTM 👍🏼

roykho commented:
Ref: https://github.com/woocommerce/woocommerce/pull/54147

✅ New logo is shown in the blocks inserter
✅ New logo is shown in the classic single product template
✅ New logo is shown in the classic checkout block

LGTM!

Konamiman commented:
@andfinally Sorry about that, https://github.com/woocommerce/woocommerce/commit/de13d0f0f855afbb62aa19c55f407ec164ac37c4 is now incorporated.

@roykho (and everybody, actually): if you have reviewed your part and you think everything is ok, please explicitly approve the pull request for easier housekeeping, thanks!

Konamiman committed:
Fix the "W" icon in the admin dashboard menu (from 53308)

elazzabi reviewed (approved):
Tested our parts, looks good!

<img width="736" alt="Screenshot 2025-01-24 at 09 14 00" src="https://github.com/user-attachments/assets/17eee4f5-9cac-42a5-8ee8-a8c17a1b2575" />
<img width="730" alt="Screenshot 2025-01-24 at 09 10 25" src="https://github.com/user-attachments/assets/5f57b8eb-b138-4542-8842-2b82bfad2b54" />

andfinally reviewed (approved):
All good for https://github.com/woocommerce/woocommerce/pull/54378, thanks! New branding appears on

- Old WCCOM connection page.
- API auth page.
- Storybook.

mcliwanow reviewed (commented):
Tested https://github.com/woocommerce/woocommerce/pull/53513 and it looks good

![Screenshot 2025-01-24 at 10 00 23](https://github.com/user-attachments/assets/615ac26b-86a0-4962-8bdd-556dd87a637c)
![Screenshot 2025-01-24 at 10 00 31](https://github.com/user-attachments/assets/b92d539e-5bed-42cd-b2d1-e48acae98579)

Aljullu reviewed (approved):
> @roykho (and everybody, actually): if you have reviewed your part and you think everything is ok, please explicitly approve the pull request for easier housekeeping, thanks!

I tested on my end too and everything looks good. So approving on behalf of WooCommerce Blocks.

KokkieH reviewed (approved):
✅ Logo updated in Storybook
✅ Legacy helper connection screen
✅ API auth page
✅ Product usage modal
✅ BFCM sale banner

Konamiman committed:
Add a data migration for the base color for emails.

The corresponding option (woocommerce_email_base_color)
gets updated to its new default value, but only if the user
hasn't set a custom color.

albarin commented:
Thanks @Konamiman.
Tested the migration of the email base color, looking good ✅

albarin reviewed (approved):
Migration of the email base color ✅

ilyasfoo commented on plugins/woocommerce/src/Internal/Admin/RemoteFreeExtensions/DefaultFreeExtensions.php:
```diff
@@ -22,9 +21,7 @@ class DefaultFreeExtensions {
-		return BrandingController::use_new_branding() ?
-			plugins_url( '/assets/images/core-profiler/woo-rebrand-2.svg', WC_PLUGIN_FILE ) :
-			plugins_url( '/assets/images/onboarding/woo.svg', WC_PLUGIN_FILE );
+		return plugins_url( '/assets/images/core-profiler/logo-woo', WC_PLUGIN_FILE );
```
Missing extension
```suggestion
		return plugins_url( '/assets/images/core-profiler/logo-woo.svg', WC_PLUGIN_FILE );
```

Konamiman replied:
Ops. Fixed, thanks.

ilyasfoo reviewed (changes requested):
Caught a minor issue, otherwise this is testing well

✅ Core profiler top logo
✅ Core profiler extensions logo when marketplace suggestion is enabled
❌ Core profiler extensions logo when marketplace suggestion is disabled
✅ LYS task success screen top logo

Konamiman committed:
Add missing .SVG extension

Co-authored-by: Ilyas Foo <foo.ilyas@gmail.com>

naman03malhotra commented on plugins/woocommerce-blocks/assets/js/icons/library/woo.tsx:
```diff
@@ -3,93 +3,57 @@
 			className={ clsx( 'woo-icon', className ) }
-			width={ width }
-			height={ height }
+			width={ 46 }
```
I might lack context, but this will always be constant from now on?

Konamiman replied:
The diff is confusing, but if you look at the file in `trunk` you'll see that the old SVG defines these "width" and "height" variables but the new one uses that value of 46 directly. You see it like that because now the markup for the new SVG is more or less in the same position in the file as the (now removed) old one was.

naman03malhotra replied:
should we then remove the variables width and height? unless they are in use.

Konamiman replied:
Hummm I'll ping @Aljullu since this change was introduced in https://github.com/woocommerce/woocommerce/pull/54147. Maybe the most sensible approach here is to leave `width={ width }` and `height={ height }` as in the old logo, I have tested that change locally and it seems to work fine. wdyt?

Aljullu replied:
Hmm, right, I left the height and width props as I thought they could override the defaults via `...props`, but I didn't realize they are not going to be in `props`. :sweat_smile: 

I can open a PR fixing this, but it shouldn't block releasing the rebranding IMO, as everywhere where we use `<Icon icon={ woo } />` we are doing so without height and width props, so it won't have any impact on the end result. Sounds good?

Aljullu replied:
I discussed this internally with @Konamiman and we decided to apply the change directly in this PR. So here is the commit: 9debc79bf564b42b4fe183704a8e1915605ef096.

naman03malhotra commented on plugins/woocommerce/client/admin/client/homescreen/mobile-app-modal/index.tsx:
```diff
@@ -43,12 +42,6 @@ export const MobileAppModal = () => {
 		ONBOARDING_STORE_NAME
 	);
 
-	if ( isNewBranding() ) {
-		import( './style-new.scss' );
-	} else {
-		import( './style-old.scss' );
-	}
```
Are the styles in `new.scss` now applied by default?

Konamiman replied:
Yes, the style that was defined differently in the "old" and "new" files has now been incorporated in `style.scss`, taking the new value.

naman03malhotra reviewed (commented):
Left some minor comments.

ilyasfoo reviewed (approved):
Thanks, core profiler LGTM!

Konamiman committed:
Remove the no longer needed BrandingController class

layoutd reviewed (approved):
👍🏻  WooCommerce Analytics install banner update OK (from https://github.com/woocommerce/woocommerce/pull/54751)

Aljullu committed:
Remove unnecessary props from Woo icon in blocks

Aljullu committed:
Update @automattic/color-studio in WooCommerce Blocks

jorgeatorres reviewed (approved):
✅ [Change the Woo logo in the marketing banner #54096](https://github.com/woocommerce/woocommerce/pull/54096)

mordeth reviewed (approved):
Moltres changes are looking good!

<img width="694" alt="Screenshot 2025-01-29 at 17 16 15" src="https://github.com/user-attachments/assets/b65a017c-97c5-498e-93d4-1ff8c26b588e" />
<img width="684" alt="Screenshot 2025-01-29 at 17 16 00" src="https://github.com/user-attachments/assets/55578865-e37e-42f9-860a-a348644c699b" />

adrianduffell reviewed (approved):
LGTM! 🚀

psealock reviewed (approved):
Changes to the Core profiler and LYS look good, new branding rendered 🚢

chris013012 commented:
Hi, the issue for attribute brands in 404 is still on 9.6.1 https://github.com/woocommerce/woocommerce/issues/54833
